### ISSUE TITLE
`await` is not treated as a plain token in a macro

### ISSUE THREAD

**Issue by kuretchi** [rust-lang/rust#57919]
```rust
macro_rules! m {
    ($($t:tt)*) => {};
}
m! { async try dyn await }
```

Compiling this code on the 2018 edition causes an following error:

```
error[E0721]: `await` is a keyword in the 2018 edition
 --> src/main.rs:4:20
  |
4 | m! { async try dyn await }
  |                    ^^^^^ help: you can use a raw identifier to stay compatible: `r#await`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0721`.
```

However, as with other keywords, this `await` keyword should be ignored through `($($t:tt)*) => {};`.

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a655ab3d2496c332f605981c221c01cb)

**Comment by varkor** [comment]
This is essentially due to https://github.com/rust-lang/rust/pull/54411.

The problem is that `await` isn't a true keyword and has been given a custom error message (triggering eagerly), which doesn't have the same handling as normal keyword use errors. It seems simplest to leave this unaddressed until the `await` syntax has been chosen and we can use `await` as a true keyword.

**Comment by dtolnay** [comment]
Fixed in 1.36.0.
